### ISSUE TITLE
🐛 flutter 2.5.0 (web) not working with cloud_firestore: ^2.5.2

### ISSUE THREAD

**Issue by jerryhamby** [FirebaseExtended/flutterfire#7006]
Looking for some help or advice.

I have created a default app I’m trying to test run, but I keep getting this error
(Only when I include cloud_firestore: ^2.5.2) otherwise
the app will launch perfectly OK. FYI, I have 26 other plugins I’m testing and
no problem except when including cloud_firestore.
Here is my test with cloud_firestore.
I have also tested cloud_firestore_web: ^2.4.2 with same results.

FYI, I have gotten 2.5.0 to work with IOS using "pod update Firebase/CoreOnly"
One more thing, using the previous version of Flutter (I think 2.2.3) everything was working great.
flutter run
```
Launching lib/main.dart on Chrome in debug mode...
Waiting for connection from debug service on Chrome...             13.4s
This app is linked to the debug service: ws://127.0.0.1:56034/noB7gMX1qaI=/ws
Debug service listening on ws://127.0.0.1:56034/noB7gMX1qaI=/ws

💪 Running with sound null safety 💪

🔥  To hot restart changes while running, press "r" or "R".
For a more detailed help message, press "h". To quit, press "q".

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:56034/noB7gMX1qaI=
TypeError: Cannot read properties of undefined (reading 'app')
    at Object.app$ [as app] (http://localhost:55985/packages/firebase_core_web/src/interop/core.dart.lib.js:31:101)
    at new cloud_firestore_web.FirebaseFirestoreWeb.new
    (http://localhost:55985/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:833:64)
    at Function.registerWith (http://localhost:55985/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:745:73)
    at Object.registerPlugins
    (http://localhost:55985/packages/flutter_application_1xxx/generated_plugin_registrant.dart.lib.js:19:46)
    at main (http://localhost:55985/web_entrypoint.dart.lib.js:31:35)
    at main.next (<anonymous>)
    at runBody (http://localhost:55985/dart_sdk.js:38659:34)
    at Object._async [as async] (http://localhost:55985/dart_sdk.js:38690:7)
    at main$ (http://localhost:55985/web_entrypoint.dart.lib.js:30:18)
    at http://localhost:55985/main_module.bootstrap.js:19:10
    at Array.forEach (<anonymous>)
    at window.$dartRunMain (http://localhost:55985/main_module.bootstrap.js:18:32)
    at <anonymous>:1:8
    at Object.runMain (http://localhost:55985/dwds/src/injected/client.js:8452:21)
    at http://localhost:55985/dwds/src/injected/client.js:24130:19
    at _wrapJsFunctionForAsync_closure.$protected (http://localhost:55985/dwds/src/injected/client.js:4461:15)
    at _wrapJsFunctionForAsync_closure.call$2 (http://localhost:55985/dwds/src/injected/client.js:11511:12)
    at Object._asyncStartSync (http://localhost:55985/dwds/src/injected/client.js:4425:20)
    at main__closure3.$call$body$main__closure (http://localhost:55985/dwds/src/injected/client.js:24142:16)
    at main__closure3.call$1 (http://localhost:55985/dwds/src/injected/client.js:24069:19)
    at StaticClosure._rootRunUnary (http://localhost:55985/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:55985/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:55985/dwds/src/injected/client.js:12826:14)
    at _ForwardingStreamSubscription._sendData$1 (http://localhost:55985/dwds/src/injected/client.js:12416:19)
    at _ForwardingStreamSubscription._add$1 (http://localhost:55985/dwds/src/injected/client.js:12362:15)
    at _ForwardingStreamSubscription._add$1 (http://localhost:55985/dwds/src/injected/client.js:12695:12)
    at _MapStream._handleData$2 (http://localhost:55985/dwds/src/injected/client.js:12756:12)
    at _ForwardingStreamSubscription._handleData$1 (http://localhost:55985/dwds/src/injected/client.js:12721:20)
    at Object.eval (eval at Closure_forwardCallTo (http://localhost:55985/dwds/src/injected/client.js:1701:14),
    <anonymous>:3:44)
    at StaticClosure._rootRunUnary (http://localhost:55985/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:55985/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:55985/dwds/src/injected/client.js:12826:14)
    at _ControllerSubscription._sendData$1 (http://localhost:55985/dwds/src/injected/client.js:12416:19)
    at _ControllerSubscription._add$1 (http://localhost:55985/dwds/src/injected/client.js:12362:15)
    at _SyncStreamController._sendData$1 (http://localhost:55985/dwds/src/injected/client.js:12201:32)
    at _SyncStreamController.add$1 (http://localhost:55985/dwds/src/injected/client.js:12082:15)
    at Object.eval (eval at Closure_forwardInterceptedCallTo
    (http://localhost:55985/dwds/src/injected/client.js:1781:14), <anonymous>:3:45)
    at StaticClosure._rootRunUnary (http://localhost:55985/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:55985/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:55985/dwds/src/injected/client.js:12826:14)
    at _ControllerSubscription._sendData$1 (http://localhost:55985/dwds/src/injected/client.js:12416:19)
    at _ControllerSubscription._add$1 (http://localhost:55985/dwds/src/injected/client.js:12362:15)
    at _SyncStreamController._sendData$1 (http://localhost:55985/dwds/src/injected/client.js:12201:32)
    at _SyncStreamController.add$1 (http://localhost:55985/dwds/src/injected/client.js:12082:15)
    at _GuaranteeSink.add$1 (http://localhost:55985/dwds/src/injected/client.js:23581:25)
    at HtmlWebSocketChannel_closure1.call$1 (http://localhost:55985/dwds/src/injected/client.js:23874:60)
    at _EventStreamSubscription_closure.call$1 (http://localhost:55985/dwds/src/injected/client.js:18042:26)
    at StaticClosure._rootRunUnary (http://localhost:55985/dwds/src/injected/client.js:4829:16)
    at _CustomZone.runUnary$2$2 (http://localhost:55985/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:55985/dwds/src/injected/client.js:12826:14)
    at _CustomZone_bindUnaryCallbackGuarded_closure.call$1 (http://localhost:55985/dwds/src/injected/client.js:13016:25)
    at invokeClosure (http://localhost:55985/dwds/src/injected/client.js:1524:26)
    at WebSocket.<anonymous> (http://localhost:55985/dwds/src/injected/client.js:1543:18)
```

**Comment by markusaksli-nc** [comment]
Hi @jerryhamby 
Have you completed the setup?
https://firebase.flutter.dev/docs/installation/web/
https://firebase.flutter.dev/docs/firestore/overview#3-web-only-add-the-sdk
Thank you

**Comment by jerryhamby** [comment]
I have looked at your links and although they are very good, right now my problem is of a much simpler nature. I’m creating a new web project in Flutter and adding just these 2 dependencies in pubspec.yaml:
1- firebase_core: ^1.6.0
2- cloud_firestore: ^2.5.2

The reason I’m going this is to test my additional dependencies, but first I need to get a very basic web app running in flutter 2.5.0. Cloud_firestore is just not working.

Adding #1 everything works OK, when I add #2 the app stops at “web_entrypoint.dart” and just sits there. Nothing prints out in the debug window. If I run “flutter run” here are the results:

An Observatory debugger and profiler on Chrome is available at: http://127.0.0.1:63200/xpfQHjDmsHk=
TypeError: Cannot read properties of undefined (reading 'app')
    at Object.app$ [as app] (http://localhost:63157/packages/firebase_core_web/src/interop/core.dart.lib.js:31:101)
    at new cloud_firestore_web.FirebaseFirestoreWeb.new
    (http://localhost:63157/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:833:64)
    at Function.registerWith (http://localhost:63157/packages/cloud_firestore_web/src/write_batch_web.dart.lib.js:745:73)
    at Object.registerPlugins (http://localhost:63157/packages/test_2_5/generated_plugin_registrant.dart.lib.js:14:46)
    at main (http://localhost:63157/web_entrypoint.dart.lib.js:31:35)
    at main.next (<anonymous>)
    at runBody (http://localhost:63157/dart_sdk.js:38659:34)
    at Object._async [as async] (http://localhost:63157/dart_sdk.js:38690:7)
    at main$ (http://localhost:63157/web_entrypoint.dart.lib.js:30:18)
    at http://localhost:63157/main_module.bootstrap.js:19:10
    at Array.forEach (<anonymous>)
    at window.$dartRunMain (http://localhost:63157/main_module.bootstrap.js:18:32)
    at <anonymous>:1:8
    at Object.runMain (http://localhost:63157/dwds/src/injected/client.js:8452:21)
    at http://localhost:63157/dwds/src/injected/client.js:24130:19
    at _wrapJsFunctionForAsync_closure.$protected (http://localhost:63157/dwds/src/injected/client.js:4461:15)
    at _wrapJsFunctionForAsync_closure.call$2 (http://localhost:63157/dwds/src/injected/client.js:11511:12)
    at Object._asyncStartSync (http://localhost:63157/dwds/src/injected/client.js:4425:20)
    at main__closure3.$call$body$main__closure (http://localhost:63157/dwds/src/injected/client.js:24142:16)
    at main__closure3.call$1 (http://localhost:63157/dwds/src/injected/client.js:24069:19)
    at StaticClosure._rootRunUnary (http://localhost:63157/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:63157/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:63157/dwds/src/injected/client.js:12826:14)
    at _ForwardingStreamSubscription._sendData$1 (http://localhost:63157/dwds/src/injected/client.js:12416:19)
    at _ForwardingStreamSubscription._add$1 (http://localhost:63157/dwds/src/injected/client.js:12362:15)
    at _ForwardingStreamSubscription._add$1 (http://localhost:63157/dwds/src/injected/client.js:12695:12)
    at _MapStream._handleData$2 (http://localhost:63157/dwds/src/injected/client.js:12756:12)
    at _ForwardingStreamSubscription._handleData$1 (http://localhost:63157/dwds/src/injected/client.js:12721:20)
    at Object.eval (eval at Closure_forwardCallTo (http://localhost:63157/dwds/src/injected/client.js:1701:14),
    <anonymous>:3:44)
    at StaticClosure._rootRunUnary (http://localhost:63157/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:63157/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:63157/dwds/src/injected/client.js:12826:14)
    at _ControllerSubscription._sendData$1 (http://localhost:63157/dwds/src/injected/client.js:12416:19)
    at _ControllerSubscription._add$1 (http://localhost:63157/dwds/src/injected/client.js:12362:15)
    at _SyncStreamController._sendData$1 (http://localhost:63157/dwds/src/injected/client.js:12201:32)
    at _SyncStreamController.add$1 (http://localhost:63157/dwds/src/injected/client.js:12082:15)
    at Object.eval (eval at Closure_forwardInterceptedCallTo
    (http://localhost:63157/dwds/src/injected/client.js:1781:14), <anonymous>:3:45)
    at StaticClosure._rootRunUnary (http://localhost:63157/dwds/src/injected/client.js:4823:18)
    at _CustomZone.runUnary$2$2 (http://localhost:63157/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:63157/dwds/src/injected/client.js:12826:14)
    at _ControllerSubscription._sendData$1 (http://localhost:63157/dwds/src/injected/client.js:12416:19)
    at _ControllerSubscription._add$1 (http://localhost:63157/dwds/src/injected/client.js:12362:15)
    at _SyncStreamController._sendData$1 (http://localhost:63157/dwds/src/injected/client.js:12201:32)
    at _SyncStreamController.add$1 (http://localhost:63157/dwds/src/injected/client.js:12082:15)
    at _GuaranteeSink.add$1 (http://localhost:63157/dwds/src/injected/client.js:23581:25)
    at HtmlWebSocketChannel_closure1.call$1 (http://localhost:63157/dwds/src/injected/client.js:23874:60)
    at _EventStreamSubscription_closure.call$1 (http://localhost:63157/dwds/src/injected/client.js:18042:26)
    at StaticClosure._rootRunUnary (http://localhost:63157/dwds/src/injected/client.js:4829:16)
    at _CustomZone.runUnary$2$2 (http://localhost:63157/dwds/src/injected/client.js:12879:39)
    at _CustomZone.runUnaryGuarded$1$2 (http://localhost:63157/dwds/src/injected/client.js:12826:14)
    at _CustomZone_bindUnaryCallbackGuarded_closure.call$1 (http://localhost:63157/dwds/src/injected/client.js:13016:25)
    at invokeClosure (http://localhost:63157/dwds/src/injected/client.js:1524:26)
    at WebSocket.<anonymous> (http://localhost:63157/dwds/src/injected/client.js:1543:18)
Activating Dart DevTools...                                      1,126ms

**Comment by markusaksli-nc** [comment]
This is because you need to complete the setup and add the CDN imports in your `index.html` for the plugins to work on web. They are not designed to work out of the box without any further setup. If you think it shouldn't at least throw an error if you haven't done the setup then there is already an open issue for this https://github.com/FirebaseExtended/flutterfire/issues/3962
